app/assets/javascripts/discourse/app/components/search-menu.js

```diff
@@ -210,6 +217,17 @@ export default class SearchMenu extends Component {
     this.search.inTopicContext = false;
   }
 
+  @action
+  cancelSearchMobile() {
+    this.close();
+
+    // clears search value if present
```

Comment:
niptpick, but IMO this is an unnecessary comment, it's quite clear what's going on